components/browser/session/src/main/java/mozilla/components/browser/session/Session.kt

```diff
@@ -239,29 +239,15 @@ class Session(
     /**
      * Set when a load request is received, indicating if the request came from web content, or via a redirect.
      */
-    var loadRequestMetadata: Consumable<LoadRequestMetadata>
-        by Delegates.vetoable(Consumable.empty()) { _, _, request ->
-            if (request.peek() == null) {
-                false
-            } else {
-                val new = request.peek() as LoadRequestMetadata
-                val consumers = wrapConsumers<LoadRequestMetadata> {
-                    onLoadRequest(
-                        this@Session,
-                        new.url,
-                        new.isSet(LoadRequestOption.REDIRECT),
-                        new.isSet(LoadRequestOption.WEB_CONTENT)
-                    )
-                }
-
-                val consumed = request.consumeBy(consumers)
-                when (consumed) {
-                    true -> new.allowOrDeny(false)
-                    false -> new.allowOrDeny(true)
-                }
-
-                !consumed
-            }
+    var loadRequestMetadata: LoadRequestMetadata by Delegates.observable(LoadRequestMetadata.blank) { _, _, new ->
```

Comment:
It looks like there's no consumer of `loadRequestMetadata` on the session anymore. Can we just remove it now incl. the correponding code in `EngineObserver`?

Reply:
Hm looks like Fenix added an observer directly on `Session`: https://github.com/mozilla-mobile/fenix/blob/d1b134c2ff33002303734f14f45c3e5a41290c74/app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt#L386

:(